testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxrs/spec/basic/DemoAppTwoTest.java

```diff
@@ -0,0 +1,92 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.jaxrs.spec.basic;
+
+import java.net.URL;
+import javax.ws.rs.core.Response;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.test.integration.jaxrs.spec.basic.resource.DemoAppResourceTwo;
+import org.jboss.as.test.integration.jaxrs.spec.basic.resource.DemoAppTwo;
+import org.jboss.as.test.integration.jaxrs.spec.basic.resource.DemoAppResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+
+/**
+ * @tpSubChapter Configuration
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Test for usage of more application in one deployment
+ * @tpSince RESTEasy 3.0.16
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class DemoAppTwoTest {
+    @ArquillianResource
+    URL baseUrl;
+
+    private static final String CONTENT_ERROR_MESSAGE = "Wrong content of response";
+
+    @Deployment
+    public static Archive<?> deploySimpleResource() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, DemoAppTwoTest.class.getSimpleName() + ".war");
+        war.addAsWebInfResource(DemoAppTwoTest.class.getPackage(), "DemoAppTwoWeb.xml", "web.xml");
+        war.addClasses(DemoAppResource.class, DemoAppResourceTwo.class,
+            DemoAppTwo.class);
+        return war;
+    }
+
+    /**
+     * The jaxrs 2.0 spec says that when a Application subclass returns
+     * values for getClasses or getSingletons methods the returned classes
+     * should be the only ones used.
+     * This test confirms that returning a value for getClasses is
+     * handled properly by the server.
+     */
+    @Test
+    public void testDemo() throws Exception {
+        // check the returned resource class is available
+        Client client = ClientBuilder.newClient();
+        String url = baseUrl.toString() + "resources";
+        WebTarget base = client.target(url);
+        String value = base.path("example").request().get(String.class);
+        Assert.assertEquals(CONTENT_ERROR_MESSAGE, "Hello world!", value);
+        client.close();
```

Comment:
this should probably be in finally block, so we always close the client even if assertion fails.

same applies for all other use cases.